workspace/service.py

```
@@ -1,31 +1,42 @@
+import be_utilities.service_exceptions as service_exceptions
 import workspace.dao as ws_dao
 
+import user.service as user_service
+
+from db_python_util.db_exceptions import EntityNotFoundException
+
 # TODO: need to fix type annotations here
 
 def create(name: str, owner: str) -> int:
-    """Create a new workspace."""
+    if user_service.get(owner) == None:
+        raise service_exceptions.MissingException(f"The user ID {owner} does not correspond to a known user.")
     if ws_dao.get_by_name(name) is not None:
-        return -1
-
+        raise service_exceptions.AlreadyExistsException(f"The name {name} is an already existing workspace.")
+    """Create a new workspace."""
     return ws_dao.create(name, owner)
 
-def update(workspace_id: int, name: str = None, owner: str = None) -> bool:
+def update(workspace_id: str, name: str = None, owner: str = None) -> bool:
+    try:
+        get(workspace_id)
+    except service_exceptions.ServiceException as e:
+        raise e
```

I think we can just let this bubble through